flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -473,8 +473,13 @@ class JobManager(
 
       val jobInfo = new JobInfo(client, listeningBehaviour, System.currentTimeMillis(),
         jobGraph.getSessionTimeout)
-
-      submitJob(jobGraph, jobInfo)
+      val jobID = jobGraph.getJobID
+      if (!currentJobs.contains(jobID)) {
```

Comment:
If you do use `submittedJobGraphs` I think you'll want to do so asynchronously in order to not block the actor.